cmd/argocd/commands/app.go

```diff
@@ -99,7 +98,7 @@ func NewApplicationAddCommand(clientOpts *argocdclient.ClientOptions) *cobra.Com
 			}
 			conn, appIf := argocdclient.NewClientOrDie(clientOpts).NewApplicationClientOrDie()
 			defer util.Close(conn)
-			_, err := appIf.Create(context.Background(), &app)
+			_, err := appIf.Create(DefaultClientContext(clientOpts), &app)
```

Comment:
Wrapping the context in every place where we make an gRPC call isn't the right approach. There is a common function in apiclient.go, NewConn() where you can adjust the DialOptions when establishing the connection. There is where we would want to add another dial option from grpc.WithPerRPCCredentials() with the token. 

See this example:
https://github.com/grpc/grpc-go/issues/106#issuecomment-246978683